### ISSUE TITLE
Change of Checkbox Options Will Keep Old Options from Remaining Selected

### ISSUE THREAD

**Issue by 3rdMillWeb** [valendesigns/option-tree#567]
If I have a set of option tree checkboxes (for this example, options 1-5) for a Post meta-box, and I select option 2 and 4 and update the article.

Because of circumstances, at some point option 1 is no longer relevant, so I remove it and now there are only options 2-4 (in a checkbox field).

When I go back to that Post that originally had options 2 and 4 checked, they will no longer be checked. They will still be in the database as a serialized array, but since the first checkbox is removed in the set, it changes the numbered keys in the serialized array and now it no longer knows to keep option 2 and 4 checked. At this point, if I open the Post, don't touch the checkboxes, and Update the Post, now the Post will not have any of the Checkboxes marked as selected in the database.

Does that make sense? Is there a way to fix this problem? I have a few sets of Checkbox fields in the Option Tree meta-boxes in which the options to Check often change, so this working correctly would make a drastic difference.

**Comment by valendesigns** [comment]
Yes, it makes sense. Fixing it though is not as straight forward. Can you create some example code that demos the issue so I can test it?

**Comment by 3rdMillWeb** [comment]
```php
add_action( 'admin_init', 'wpl_meta_boxes' );

function wpl_meta_boxes() {

  $short_term_opportunities_sort_box = array(
    'id'       => 'short_term_opportunities_sort_box',
    'title'     => 'Short-term Opportunity Sort Fields',
    'desc'      => '',
    'pages'     => array( 'post' ),
    'context'    => 'normal',
    'priority'   => 'high',
    'fields'    => array(
      array(
        'label'     => 'Sort Duration',
        'id'       => 'sort_duration',
        'type'      => 'checkbox',
        'desc'      => 'Select ALL that apply',
        'choices'    => array (
          array(
            'value'     => '2-6-weeks',
            'label'     => __( '2 - 6 Weeks', 'text-domain' ),
          ),
          array(
            'value'     => '6-12-weeks',
            'label'     => __( '6 - 12 Weeks', 'text-domain' ),
          ),
          array(
            'value'     => '3-6-months',
            'label'     => __( '3 - 6 Months', 'text-domain' ),
          ),
          array(
            'value'     => '6-months',
            'label'     => __( '6 Months +', 'text-domain' ),
          ),
        );
      );
    )
  );
  ot_register_meta_box( $short_term_opportunities_sort_box );

}
```
This is a simple piece of static checkboxes that I have using OT. I do have more dynamic checkbox sets that pulling in post lists and such, but this example will suffice.

In my example shown above, if I go to a 'post' and check the first couple of values ('2-6-weeks' and '6-12-weeks'), then at some point in time, if I add another option to this list of checkboxes in this PHP file, and change the list so that the end of it looks like this:
```php
          array(
            'value'     => '6-months',
            'label'     => __( '6 Months - 12 Months', 'text-domain' ),
          ),
          array(
            'value'     => '12-months',
            'label'     => __( '1 Year +', 'text-domain' ),
          )
```
After I make this change and go back to the database, '6-12-weeks' and '2-6-weeks' are saved in the database as a serialized array. However, if I go back to the 'post', the ones that I have checked (as shown by the serialized array in the database), will not display as being checked. Then if I save the 'post', it will overwrite the database entry with the current status of the checkboxes (which are that none are checked), and the post will no longer have any checked items.
